### ISSUE TITLE
403 Client Error

### ISSUE THREAD

**Issue by Hood-fr** [AlexandrErohin/home-assistant-flightradar24#73]
My FR24 integration had been working perfectly for a week
And since this morning, I have either the "403 Client Error: Forbidden for url: https://www.flightradar24.com/user/login" error when I try to configure using my FR24 account, or nothing in the sensors when I configure without my credentials

Any ideas ?

**Comment by ohuc** [comment]
same!

**Comment by AlexandrErohin** [comment]
@Hood-fr 
1. Can you enter your FR24 account in https://www.flightradar24.com?
2. Is there any errors when you use integration without your FR24 account ?

**Comment by Hood-fr** [comment]
1. Yes no issue accessing/ login into FR24 through the web browser, even address https://data-cloud.flightradar24.com/zones/fcgi/feed.js?... is returning a result when put into the web browser
2. No error pops up, put the sensor are all empty, no flight inside even if you see some A/C flying through your zone in FR24

Could it be some banning by Cloudflare or equivalent ? https://community.cloudflare.com/t/403-on-iot/675145

**Comment by Hood-fr** [comment]
It works from my PC but I don't know how to test that from the same platform that my HA instance is working on (Docker container in a Synology NAS)

**Comment by Hood-fr** [comment]
> same!

Out of curiosity, on what platform is your HomeAssistant running ?

**Comment by AlexandrErohin** [comment]
Do you use AdGuard or something similar? AdGuard can block the connection. Look here
https://github.com/AlexandrErohin/home-assistant-flightradar24/issues/26#issuecomment-2017252622

**Comment by Hood-fr** [comment]
No I don't

**Comment by ohuc** [comment]
> > same!
> 
> Out of curiosity, on what platform is your HomeAssistant running ? @ohuc

Rpi 4 homeassistnt OS

**Comment by ohuc** [comment]
This is what I've been getting in my logs
`
This error originated from a custom integration.

Logger: custom_components.flightradar24
Source: custom_components/flightradar24/coordinator.py:134
integration: Flightradar24 (documentation, issues)
First occurred: October 12, 2024 at 4:48:23 PM (150173 occurrences)
Last logged: 8:38:51 AM

403 Client Error: Forbidden for url: https://data-cloud.flightradar24.com/zones/fcgi/feed.js?faa=1&satellite=1&mlat=1&flarm=1&adsb=1&gnd=1&air=1&vehicles=1&estimated=1&maxage=14400&gliders=1&stats=1&limit=5000&bounds=22.57429817424915%2C22.502352452944496%2C88.31799231166158%2C88.39588742312442
403 Client Error: Forbidden for url: https://data-cloud.flightradar24.com/zones/fcgi/feed.js?faa=1&satellite=1&mlat=1&flarm=1&adsb=1&gnd=1&air=1&vehicles=1&estimated=1&maxage=14400&gliders=1&stats=1&limit=5000&bounds=22.54732292313755%2C22.529336491131186%2C88.34719346637429%2C88.36666724101487
403 Client Error: Forbidden for url: https://data-cloud.flightradar24.com/zones/fcgi/feed.js?faa=1&satellite=1&mlat=1&flarm=1&adsb=1&gnd=1&air=1&vehicles=1&estimated=1&maxage=14400&gliders=1&stats=1&limit=5000&bounds=22.541027938491812%2C22.535632008859324%2C88.35400871036356%2C88.35985084269703`

**Comment by Hood-fr** [comment]
OK got some news:

Tried to send some HEAD request to flightradar24 via curl from my PC and my Synology and this happened:
My PC:
```
curl -v -I https://www.flightradar24.com/user/login/
* Host www.flightradar24.com:443 was resolved.
* IPv6: 2606:4700::6812:6170, 2606:4700::6812:6270
* IPv4: 104.18.98.112, 104.18.97.112
*   Trying [2606:4700::6812:6170]:443...
* Connected to www.flightradar24.com (2606:4700::6812:6170) port 443
* * ALPN: curl offers h2,http/1.1
[TLS handshake sequence removed]
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=flightradar24.com
*  start date: Oct  8 05:43:09 2024 GMT
*  expire date: Jan  6 05:43:08 2025 GMT
*  subjectAltName: host "www.flightradar24.com" matched cert's "*.flightradar24.com"
*  issuer: C=US; O=Google Trust Services; CN=WE1
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://www.flightradar24.com/user/login/
* [HTTP/2] [1] [:method: HEAD]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: www.flightradar24.com]
* [HTTP/2] [1] [:path: /user/login/]
* [HTTP/2] [1] [user-agent: curl/8.7.1]
* [HTTP/2] [1] [accept: */*]
> HEAD /user/login/ HTTP/2
> Host: www.flightradar24.com
> User-Agent: curl/8.7.1
> Accept: */*
> 
* Request completely sent off
< HTTP/2 200 
HTTP/2 200 
```
My NAS:
```
curl -v -I https://www.flightradar24.com/user/login/
* Host www.flightradar24.com:443 was resolved.
* IPv6: 2606:4700::6812:6270, 2606:4700::6812:6170
* IPv4: 104.18.97.112, 104.18.98.112
*   Trying [2606:4700::6812:6270]:443...
* Immediate connect fail for 2606:4700::6812:6270: Network is unreachable
*   Trying [2606:4700::6812:6170]:443...
* Immediate connect fail for 2606:4700::6812:6170: Network is unreachable
*   Trying 104.18.97.112:443...
* ALPN: curl offers h2,http/1.1
[TLS handshake sequence removed]
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=flightradar24.com
*  start date: Oct  8 05:43:09 2024 GMT
*  expire date: Jan  6 05:43:08 2025 GMT
*  subjectAltName: host "www.flightradar24.com" matched cert's "*.flightradar24.com"
*  issuer: C=US; O=Google Trust Services; CN=WE1
* Connected to www.flightradar24.com (104.18.97.112) port 443
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://www.flightradar24.com/user/login/
* [HTTP/2] [1] [:method: HEAD]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: www.flightradar24.com]
* [HTTP/2] [1] [:path: /user/login/]
* [HTTP/2] [1] [user-agent: curl/8.10.1]
* [HTTP/2] [1] [accept: */*]
> HEAD /user/login/ HTTP/2
> Host: www.flightradar24.com
> User-Agent: curl/8.10.1
> Accept: */*
< HTTP/2 403 
HTTP/2 403 
```

I've realized that my PC was connecting through IPv6, while I deactivated IPv6 on my NAS. My NAS was connecting through IPv4, was handshaking TLS, and then got rejected. 

I reactivated IPv6 on my NAS and this I am now back in business, works like a charm... for now

I still believe that CloudFare somehow banned my IPv4 connection at some point (too many API requests? Is there a limit) and I now hope that I don't get banned on IPv6

**Comment by AlexandrErohin** [comment]
Great that you found the reason and fixed it!
@ohuc Have you tried this?

**Comment by Hood-fr** [comment]
By the way, @AlexandrErohin you mention a link to /edit-configuration/ (relative to adding tracked flights) in the main page but the link is dead.

**Comment by ohuc** [comment]
> Great that you found the reason and fixed it! @ohuc Have you tried this?

nope, i am running homeassistant os when i ran the curl command `curl -v -I https://www.flightradar24.com/user/login/` i get http/2 200 response but the flightradar24 integration still shows the same error

**Comment by AlexandrErohin** [comment]
@ohuc You run the curl command inside HA OS? or in your PC?

**Comment by ohuc** [comment]
In ha Is Terminal

**Comment by AlexandrErohin** [comment]
@ohuc Have you tried this client https://github.com/JeanExtreme002/FlightRadarAPI/tree/main/python?
```
from FlightRadar24 import FlightRadar24API
fr_api = FlightRadar24API(user='username', password='password')
``

**Comment by AlexandrErohin** [comment]
As this integration uses FlightRadarAPI client - it is better to post there the problem with details. And I have found there the issue already opened JeanExtreme002/FlightRadarAPI#88

My integration is still working, so i cannot debug it

**Comment by fifistargate** [comment]
Didnt changed anything and since this morning is working again..

**Comment by Hood-fr** [comment]
Same here. Lost it for a couple of days an now it is back
